Modules/mathmodule.c

```diff
@@ -2016,6 +2016,71 @@ math_factorial(PyObject *module, PyObject *arg)
 }
 
 
+/*[clinic input]
+math.lcm
+
+    x as a: object
+    y as b: object
+    /
+least common multiple of x and y
+[clinic start generated code]*/
+
+static PyObject *
+math_lcm_impl(PyObject *module, PyObject *a, PyObject *b)
+/*[clinic end generated code: output=6f83fb6d671074ba input=bd41b785dc2a4ff1]*/
+{
+    PyObject *g, *m, *f, *ab;
+    
+    a = PyNumber_Index(a);
```

Comment:
Like most (but not all) Python C-API functions, `PyNumber_Index` returns a new reference. So at this point, we own the reference to `a`, and we're responsible to making sure it's eventually DECREF'd in all code paths below. *Except*: if `PyNumber_Index` failed, then `a` is `NULL` and we don't own any references.

Owned references before this line: None
Owned references after this line: `a` if `PyNumber_Index` succeeded; None if it failed.